### PR TITLE
latino 1.3.0

### PR DESCRIPTION
**Pull request by ChannelJuanNews**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

brew formula for the `latino` programming language (programming language in Spanish) officially known as [Lenguaje Latino](https://www.lenguajelatino.org/)

### CODE REVIEW

**Comment by ChannelJuanNews** [comment]
This is a retry since my previous pull request was not up to date with homebrew-core so it had over 5000 file changes. This only changes one file which is the creation of the `latino.rb` formula.

File: Formula/latino.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    depends_on "make" => :build
+  end
+  fails_with :clang do
+    build 600
+    cause "multiple configure and compile errors"
+  end
+  fails_with :gcc do
+    build 600
+    cause "multiple configure and compile errors"
+  end
```
Comment (carlocab) [inline Formula/latino.rb:23]:
```suggestion
  url "https://github.com/lenguaje-latino/Latino/archive/v1.3.0.tar.gz"
  sha256 "28118a047265394ad9a5fcb2c76895d08af4b43a5886f52afce98a0f8cdc86fe"
  license "MIT"
  head "https://github.com/lenguaje-latino/Latino.git"
  
  depends_on "cmake" => :build
  depends_on "make" => :build
```

Will this build with the built-in macOS `make`? If it does, then you don't need to declare `depends_on "make"`.

I think we can probably also skip the `fails_with` blocks.

File: Formula/latino.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  def install
+    mv "logo/Latino-logo.png", "logo/desktop.png"
```
Comment (carlocab) [inline Formula/latino.rb:25]: Why is this needed?

File: Formula/latino.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    system "cmake", ".", *std_cmake_args
+    system "make"
+    system "make", "install"
+  end
+  test do
+    system bin/"latino", "-v"
```
Comment (carlocab) [inline Formula/latino.rb:31]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/latino.rb (hunk 1 above)
Comment (ChannelJuanNews) [reply Formula/latino.rb:23]: 1. yes this will build with the built-in macOS make. 

2. okay will get rid fo depends on make 

3. okay will get rid of fails_with

Comment (carlocab) [reply Formula/latino.rb:23]: Great. You also don't need the `stable` and `head` blocks; see the suggestion in my previous comment.

File: Formula/latino.rb (hunk 2 above)
Comment (ChannelJuanNews) [reply Formula/latino.rb:25]: the tagged/stable version calls the logo `Latino-logo.png` but in the cmake.txt it references `desktop.png` which causes an error with the build. Simply re-naming it allows the build to succeed. This will not be needed once the next stable version comes out, whenever that might be

File: Formula/latino.rb (hunk 3 above)
Comment (ChannelJuanNews) [reply Formula/latino.rb:31]: Yes, here is the new and improved test 

```
  test do
    (testpath/"test1.lat").write "poner('hola mundo')"
    (testpath/"test2.lat").write "i=0\nrepetir\n\tponer(i)\n\ti++\nhasta(i>=10)"
    output = shell_output("#{bin}/latino test1.lat")
    assert_equal "hola mundo", output.chomp
    output2 = shell_output("#{bin}/latino test2.lat")
    assert_equal "0\n1\n2\n3\n4\n5\n6\n7\n8\n9", output2.chomp
  end
```

This creates 2 small `latino` programs and runs them and then compares their outputs with the expected output

**Comment by ChannelJuanNews** [comment]
I believe I made all the edits that you requested @carlocab 

Please let me know if there is anything else that you would like me to change

File: Formula/latino.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+class Latino < Formula
+  desc "Lenguaje de programación de código abierto para latinos y de habla hispana"
+  homepage "https://www.lenguajelatino.org/"
+  url "https://github.com/lenguaje-latino/Latino/archive/v1.3.0.tar.gz"
+  sha256 "28118a047265394ad9a5fcb2c76895d08af4b43a5886f52afce98a0f8cdc86fe"
+  license "MIT"
+  head "https://github.com/lenguaje-latino/Latino.git"
+  depends_on "cmake" => :build
+  depends_on "make" => :build
+  def install
```
Comment (SMillerDev) [inline Formula/latino.rb:10]: Adds some whitespace for readability and removes the dependency on make as discussed earlier

```suggestion
  head "https://github.com/lenguaje-latino/Latino.git"

  depends_on "cmake" => :build

  def install
```

File: Formula/latino.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    mv "logo/Latino-logo.png", "logo/desktop.png"
+    system "cmake", ".", *std_cmake_args
+    system "make"
+    system "make", "install"
+  end
+  test do
+    (testpath/"test1.lat").write "poner('hola mundo')"
+    (testpath/"test2.lat").write "i=0\nrepetir\n\tponer(i)\n\ti++\nhasta(i>=10)"
```
Comment (SMillerDev) [inline Formula/latino.rb:18]:
```suggestion
    (testpath/"test2.lat").write <<~EOS
      i=0
      repetir
        poner(i)
        i++
      hasta(i>=10)
    EOS
```

**Comment by ChannelJuanNews** [comment]
@SMillerDev I added your changes thank you for those suggestions

**Comment by SMillerDev** [comment]
CI is saying this is a fork. Why not point to the official repository?

**Comment by ChannelJuanNews** [comment]
@SMillerDev it is the official repository. It was forked from the original creator's personal account and added to the official organziation's repository 

original creator's repo: https://github.com/MelvinG24/Latino
official language repo https://github.com/lenguaje-latino/Latino

**Comment by SMillerDev** [comment]
It's behind the author's personal repo, which seems to be more active. I think that should be upstream.

**Comment by ChannelJuanNews** [comment]
@SMillerDev okay. Two questions. 

1. what happens if/when the changes go upstream to the official repository. Do we just update the URL in the brew formla? what is the expected action when that happens? Will it break anything?  

2. why does it matter if both repositories have the exact same official release version of `1.3.0`

**Comment by SMillerDev** [comment]
> what happens if/when the changes go upstream to the official repository. Do we just update the URL in the brew formla? what is the expected action when that happens? Will it break anything?

If the organization fork becomes more active then the original repo and the original author redirects to the organization as the main platform we can change the URL. Nothing will break in any case since brew keeps its own compiled copies.

> why does it matter if both repositories have the exact same official release version of 1.3.0

Because we want to ship the original source of the software. If Amazon forks Java we won't be shipping the Amazon version, we'll ship the original Java version.

**Comment by ChannelJuanNews** [comment]
@SMillerDev As long as changing the URLs doesn't break anything in the future, we are good to go. 

I have changed where the URLs point to. 

What are the next steps to get this merged

File: Formula/latino.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+class Latino < Formula
+  desc "Lenguaje de programación de código abierto para latinos y de habla hispana"
+  homepage "https://www.lenguajelatino.org/"
+  url "https://github.com/MelvinG24/Latino/archive/v1.3.0.tar.gz"
+  sha256 "28118a047265394ad9a5fcb2c76895d08af4b43a5886f52afce98a0f8cdc86fe"
+  license "MIT"
+  head "https://github.com/MelvinG24/Latino"
+  depends_on "cmake" => :build
+  def install
```
Comment (SMillerDev) [inline Formula/latino.rb:9]:
```suggestion
  head "https://github.com/MelvinG24/Latino"

  depends_on "cmake" => :build

  def install
```

File: Formula/latino.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    mv "logo/Latino-logo.png", "logo/desktop.png"
+    system "cmake", ".", *std_cmake_args
+    system "make"
+    system "make", "install"
+  end
+  test do
```
Comment (SMillerDev) [inline Formula/latino.rb:15]:
```suggestion
  end

  test do
```

**Review by SMillerDev — COMMENTED** [review]
Let's add some whitespace for readability

**Comment by ChannelJuanNews** [comment]
@SMillerDev now that all the checks have passed, any extra steps?

**Review by SMillerDev — APPROVED** [review]
Thanks @ChannelJuanNews  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉
